src/OrderBy.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * @template T
+ * @template U
+ * @returns {Array<U>}
+ */
+
+const _defaultComparer = function(a, b) { 
+  if(a < b) return -1;
+  else if(a > b) return 1;
+  else return 0; 
+}
+
+function OrderBy (keySelector, comparer = _defaultComparer) {
+  if (this === undefined) throw 'Undefined value exception!';
+  if (this === null && typeof this === 'object') throw 'Null value exception!';
+  return eval($.toJSON(this)) = function () {
```

Comment:
What is the intention here? jQuery is not part of this project. This will also throw a `ReferenceError` due to invalid left-hand in the assignment.

Reply:
Was supposed to address fact that original array should not be modified through the two functions. Will look for an alternate method for deep copy.

Reply:
Shallow copy is fine and actually preferred.